bf2fc6cc711aee1a0c2a/kas-fleet-manager
# chore: remove Observatorium DEX authentication support

## Description

Related to https://issues.redhat.com/browse/MGDSTRM-10990

Initially we used DEX based authentication when authenticating against Observatorium in KAS Fleet Manager and the Dataplanes.

Currently however we don't authenticate using DEX anymore but through OIDC.
Neither our Stage or Production environments uses DEX. The control plane development team does not make use of that authentication method either. The testing pipelines neither and we are not aware of anyone using it anymore.

Currently KFM code around Observatorium authentication is confusing and a maintenance burden due to the combination of OIDC and DEX. If we remove the DEX authentication support we will greatly simplify that area.

This PR takes care of that.

Specifically, the following KFM CLI flags are removed:
```
--dex-password-file
--dex-secret-file
--dex-url
--dex-username
--observatorium-gateway # This flag is only used for Observatorium DEX authentication
--observatorium-tenant # This flag is only used for Observatorium DEX authentication
--observatorium-token-file # This flag is only used for Observatorium DEX authentication
```

It also takes care of removing all the related code, OCP templates and Makefile contents that are associated/related to those flags

Additional remarks:
* This will need a post-cleanup of the KFM Stage and Production environments OCP template parameters
* This will need a post-cleanup of the KFM Stage and Production environments Vault secrets
* This needs some changes in the managed-kafka-versions repository. The changes in there should be rolled out before merging this PR. I am currently working on preparing those changes and I will notify it in this PR once that is ready
* This needs some changes in the KAS installer repository. The changes have already been made and rolled out there
* There is an additional CLI flag `--observatorium-auth-type` that with the removal of the `dex` authentication support only accepts the `redhat` value (for Red Hat SSO based auth). I intentionally left that flag only accepting one value for this PR. We can decide we either keep with the potential to be expanded later or we simply remove it to simplify further that area. I would like a discussion about it but let's do it through a separate channel and independently of this PR
* I have identified other flags that might be further removed but unrelated to DEX. I will open a different PR for that.

## Verification Steps

First of all, make sure in your local environment the following local secrets are removed:
```
secrets/dex.secret
secrets/dex.password
secrets/observatorium.token
```

* KAS Fleet Manager can run correctly
* Tests pass


## Checklist (Definition of Done)
